sharding/peers.go

```diff
@@ -33,6 +35,46 @@ type peer struct {
 	address string
 }
 
+func (p *Peers) SmallestAvailableShardID() (string, error) {
+	peerList := make([]int, len(p.peers), len(p.peers))
```

Comment:
You can also just do `make([]int, 0, len(p.peers))` and `peerList = append(peerList, num)` so you don't need the `i` counter, but meh.